### PR TITLE
Add restriction: lwt < 4.0.0

### PR DESCRIPTION
**Pull request by akabe**
ocaml-jupyter kernel does not support lwt 4.0.0+ (will be released in March 2018) due to changes of `lwt.ppx` and `lwt.log`: https://github.com/ocsigen/lwt/issues/453
I plan to upgrade `jupyter` package for supporting the new lwt after its release.

@aantron Thanks for notification of the breaking changes!

This PR cannot pass OPAM CI perfectly:

- `archimedes` package cannot be installed in some Docker images because they do not have `Graphics` package.
- opam packages containing `source` section (like `zmq`) cannot be installed by OPAM 2 since `source` is deprecated.

### CODE REVIEW

**Comment by aantron** [comment]
@akabe, you can actually make all releases of `jupyter` compatible with all past Lwt, and all future Lwt until we stop supporting `lwt_log`, by making it depend on `lwt_ppx` and `lwt_log`, which are two new opam packages. You don't need any constraints on Lwt.

`lwt_ppx` and `lwt_log` have/will have constraints on Lwt maintained by us (Lwt). So, when `lwt_log` becomes too out-of-date to keep up with Lwt, we will contrain it, and the effect will automatically propagate to `jupyter`.

You can depend on these packages from old `jupyter` releases. The current `lwt_ppx` and `lwt_log` are basically aliases for "any version of Lwt" at the moment.

In summary, you just add dependencies on `lwt_ppx` and `lwt_log`, and you don't have to do anything else ever again, unless depending on `lwt_log` one day makes you unable to use a then-modern Lwt. Then, you port to `Logs_lwt` or some other logging method, and continue :)

**Comment by aantron** [comment]
Ah sorry, you're right, if you mention `lwt.ppx`, then you have to constrain `lwt` to less than 4.0.0. Forget my comment :p

**Comment by aantron** [comment]
And thanks for constraining releases of `jupyter` :)
